pytypes/type_util.py

```diff
@@ -216,16 +216,19 @@ def get_orig_class(obj, default_to__class__=False):
             cls = obj.__class__
         if _typing_3_7 and is_Generic(cls):
             # Workaround for https://github.com/python/typing/issues/658
-            stck = stack()
+            frame = currentframe()
             # Searching from index 2 is sufficient: At 0 is get_orig_class, at 1 is the caller.
             # We assume the caller is not typing._GenericAlias.__call__ which we are after.
-            for line in stck[2:]:
+            while frame.f_back.f_back:
                 try:
-                    res = line[0].f_locals['self']
+                    res = frame.f_locals['self']
                     if res.__origin__ is cls:
                         return res
                 except (KeyError, AttributeError):
-                    pass
+                    old_frame = frame
+                    frame = frame.f_back
+                    del old_frame
```

Comment:
I believe this does not really delete the reference to ``frame`` and that it is not necessary OTOH to attempt a deletion in every loop cycle. I think one ``del frame`` right before every possible function exit should be sufficient and ideal.
I.e. right before the ``return`` in
```
if res.__origin__ is cls:
   return res
```
plus right after the loop exits (in failure case).